### ISSUE TITLE
Support option that only checks the php files that you changed

### ISSUE THREAD

**Issue by paladox** [JakubOnderka/PHP-Parallel-Lint#75]
Hi please could you support only checking the php files you changed in an option please?

**Comment by grogy** [comment]
Hello,

how you can detect changed files?

Thanks

**Comment by paladox** [comment]
We could use git, see how Wikimedia do it https://github.com/wikimedia/integration-config/blob/master/jjb/macro.yaml#L541 and https://github.com/wikimedia/integration-jenkins/blob/master/bin/git-changed-in-head#L42 please.

We need to do that for mediawiki due to the amout of files.

**Comment by grogy** [comment]
Yes, it is good idea. But.. what if project don't use git? It is next dependency (for that option).

So - is not a solution use `--exclude` for ignore libraries? Next good option is `-j` - it can run a linter in more threads. What do you mean about it?

**Comment by paladox** [comment]
Well we could support a new option and call it like -h probably, and write in the README that this option requires git. So we doint need to add dependency on anything but just put a not a read me file explaning that you need to to run -h.

**Comment by grogy** [comment]
So ok.. can you send PR? I can help you with code-review and I can help with design changes ;-)

**Comment by paladox** [comment]
Oh, but I'm not sure how to do this.

**Comment by grogy** [comment]
Start point for you will be:

1/ configuration for options - https://github.com/JakubOnderka/PHP-Parallel-Lint/blob/master/src/Settings.php#L125
2/ process runner - https://github.com/JakubOnderka/PHP-Parallel-Lint/blob/master/src/Manager.php#L46

It is better now?

**Comment by paladox** [comment]
Yep, thanks. I will see if I can do it now :)

**Comment by willemstuursma** [comment]
I don't think this tool should do this, but it's the job of the task runner. See #58.

**Comment by flow-control** [comment]
You could use [grumphp](https://github.com/phpro/grumphp) to achieve this

**Comment by exussum12** [comment]
This option is already supported ?

git diff origin/master  --name-only -- '*.php' | parallel-lint --stdin

Will use git as your VCS to tell parallel-lint which files to look for ?
